tests/rptest/tests/adjacent_segment_merging_test.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Licensed as a Redpanda Enterprise file under the Redpanda Community
+# License (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+# https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import RedpandaService, SISettings
+
+from rptest.clients.types import TopicSpec
+from rptest.clients.rpk import RpkTool
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.util import (
+    wait_until, )
+
+from ducktape.mark import matrix
+
+import time
+import os
+import json
+
+from rptest.utils.si_utils import gen_segment_name_from_meta
+
+# Log errors expected when connectivity between redpanda and the S3
+# backend is disrupted
+CONNECTION_ERROR_LOGS = [
+    # e.g. archival - [fiber1] - service.cc:484 - Failed to upload 3 segments out of 4
+    r"archival - .*Failed to upload \d+ segments"
+]
+
+
+class AdjacentSegmentMergingTest(RedpandaTest):
+    s3_topic_name = "panda-topic"
+    topics = (TopicSpec(name=s3_topic_name,
+                        partition_count=1,
+                        replication_factor=3), )
+
+    def __init__(self, test_context):
+        si_settings = SISettings(
+            cloud_storage_max_connections=10,
+            log_segment_size=0x10000,
+            cloud_storage_segment_max_upload_interval_sec=1,
+            cloud_storage_enable_remote_write=True)
+
+        xtra_conf = dict(
+            cloud_storage_housekeeping_interval_ms=10000,
+            cloud_storage_idle_timeout_ms=200,
+            cloud_storage_segment_size_target=1024 * 1024 * 10,
+            cloud_storage_segment_size_min=1024 * 1024 * 8,
+        )
+
+        self.s3_bucket_name = si_settings.cloud_storage_bucket
+
+        super(AdjacentSegmentMergingTest,
+              self).__init__(test_context=test_context,
+                             extra_rp_conf=xtra_conf,
+                             si_settings=si_settings)
+
+        self.kafka_tools = KafkaCliTools(self.redpanda)
+        self.rpk = RpkTool(self.redpanda)
+
+    def setUp(self):
+        super().setUp()  # topic is created here
+
+    def tearDown(self):
+        self.s3_client.empty_bucket(self.s3_bucket_name)
```

Comment:
The name of `s3_client` changed to `cloud_storage_client` to support Azure parametrisation.